Tests/kaas/k8s-version-recency/k8s-version-recency-check.py

```diff
@@ -0,0 +1,339 @@
+        "Authorization": f"Bearer {token}",
+        "X-GitHub-Api-Version": "2022-11-28"
+    }
+
+    # Request latest version
+    resp = requests.get("https://api.github.com/repos/kubernetes/kubernetes/releases/latest",
+                        headers=github_headers).json()
+
+    # Latest version
+    lv = K8sVersionInfo()
+    lv.extract_version(resp['tag_name'].strip("v"))
+    lv.date = datetime.datetime.strptime(resp['published_at'], '%Y-%m-%dT%H:%M:%SZ')
+
+    # Request the latest 100 version (the next are not needed, since these versions are too old)
+    resp = requests.get("https://api.github.com/repos/kubernetes/kubernetes/releases?per_page=100",
+                        headers=github_headers).json()
+
+    # Iterate all versions until the first patch of the previous latest version
+    for i in range(0, len(resp)):
+        v = K8sVersionInfo()
+        v.extract_version(resp[i]['tag_name'].split("-")[0].strip("v"))
+        v.date = datetime.datetime.strptime(resp[i]['published_at'], '%Y-%m-%dT%H:%M:%SZ')
+
+        if resp[i]['draft'] is False and resp[i]['prerelease'] is False:
+            k8s_versions.append(v)
+
+        # Stop adding new version if the version we added is the previous latest minor versions first patch,
+        # since it is most of the time around 3-6 months until a new version comes out or an old one goes into EOL
+        if v.check_for_version(lv.major, (lv.minor-1), 0):
+            break
+
+    # Filter the versions depending on the usage times set by the standard
+    for i in range(0, len(k8s_versions)):
+        v = k8s_versions.__getitem__(i)
+        try:
+            if diff_months(v.date, datetime.datetime.now()) >= MINOR_VERSION_CADENCE:
+                v = None
+            else:
+                for kv in k8s_versions:
+                    if kv is not None and \
+                       v.check_for_version(major=kv.major, minor=kv.minor) and v.patch < kv.patch:
+                        if diff_weeks(datetime.datetime.now(), kv.date) >= PATCH_VERSION_CADENCE:
+                            v = None
+                            break
+                        if kv in cve_versions and \
+                           diff_days(datetime.datetime.now(), kv.date) >= CVE_VERSION_CADENCE:
+                            v = None
+                            break
+        except (KeyError, IndexError, TypeError) as e:
+            logger.debug(f"An error occurred during version filtering: {e}")
+            v = None
+        finally:
+            k8s_versions[i] = v
+
+    return [v for v in k8s_versions if v is not None]
+
+
+async def main(argv):
+    global return_code
+
+    try:
+        config = load_config(argv)
+    except ConfigException:
+        return_code = 1
+        return
+
+    cve_versions = check_cve_versions()
+    k8s_versions = get_k8s_versions(config.token, cve_versions)
+    cluster_version, cluster_name = await get_k8s_cluster_version(config.kubeconfig)
```

Comment:
Why not get the cluster version first and then, knowing the current version, look for CVEs and Kubernetes releases? Then you don't have to fetch all CVEs and all releases, only to restrict your attention later. Am I missing something?

Reply:
That can be done, but like I asked in one of the first comments (and this was also an intention here), you could test multiple clusters with this script, if we want the functionality for this.
Yeah we could still do that, but that were my intentions with this order.

Reply:
Got it.